File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/scene/ArcgisSceneViewFactory.java
package com.valentingrigorean.arcgis_maps_flutter.scene;

import android.content.Context;

import com.valentingrigorean.arcgis_maps_flutter.LifecycleProvider;

import java.util.Map;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class ArcgisSceneViewFactory extends PlatformViewFactory {

    private final BinaryMessenger binaryMessenger;
    private final LifecycleProvider lifecycleProvider;

    public ArcgisSceneViewFactory(BinaryMessenger binaryMessenger, LifecycleProvider lifecycleProvider) {
        super(StandardMessageCodec.INSTANCE);
        this.binaryMessenger = binaryMessenger;
        this.lifecycleProvider = lifecycleProvider;
    }

    @Override
    public PlatformView create(Context context, int viewId, Object args) {
        Map<String, Object> params = (Map<String, Object>) args;
        return new ArcgisSceneController(viewId, context, params, binaryMessenger, lifecycleProvider);
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/scene/ArcgisSceneViewFactory.kt
package com.valentingrigorean.arcgis_maps_flutter.scene

import android.content.Context
import androidx.lifecycle.Lifecycle
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class ArcgisSceneViewFactory(
    private val binaryMessenger: BinaryMessenger,
    private val lifecycleProvider: () -> Lifecycle
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val params = args as Map<String?, Any?>?
        return ArcgisSceneController(viewId, context, params, binaryMessenger, lifecycleProvider)
    }
}